File: app/models/source_breakdown.py
from app.extensions import db


class SourceBreakdown(db.Model):
    __tablename__ = "source_breakdowns"

    id = db.Column(db.Integer, primary_key=True)
    snapshot_id = db.Column(db.Integer, db.ForeignKey("snapshots.id", ondelete="CASCADE"), nullable=False)
    department = db.Column(db.String(50), nullable=False)
    source = db.Column(db.String(100), nullable=False)
    amount = db.Column(db.Integer)
    pct_of_gifts = db.Column(db.Numeric(8, 6))

    def __repr__(self):
        return f"<SourceBreakdown {self.department}/{self.source}>"

File: src/models/sourceBreakdown.js
const { DataTypes } = require('sequelize');

module.exports = (sequelize) => {
  return sequelize.define('SourceBreakdown', {
    id: { type: DataTypes.INTEGER, primaryKey: true, autoIncrement: true },
    snapshotId: { type: DataTypes.INTEGER, allowNull: false, field: 'snapshot_id' },
    department: { type: DataTypes.STRING(50), allowNull: false },
    source: { type: DataTypes.STRING(100), allowNull: false },
    amount: { type: DataTypes.INTEGER },
    pctOfGifts: { type: DataTypes.DECIMAL(12, 6), field: 'pct_of_gifts' },
  }, {
    tableName: 'source_breakdowns',
    timestamps: false,
  });
};
